src/cedict-service/service.js

```diff
@@ -0,0 +1,34 @@
+import MessagingService from '@lexisCs/messaging/messaging-service.js'
+import ResponseMessage from '@lexisCs/messaging/messages/response-message.js'
+import Destination from '@lexisCs/messaging/destinations/window-iframe-destination.js'
+import CedictData from '@lexisCs/cedict-service/cedict-data.js'
+import CedictSchema from '@lexisCs/schemas/cedict.js'
+
+let cedictData
+
+const messageHandler = (request, responseFn) => {
+  let response
+  if (!cedictData.isReady) {
+    responseFn(ResponseMessage.Error(request, new Error('Uninitialized')))
+    return
+  }
+
+  if (request.body.getWords) {
```

Comment:
Here is how we're analyzing message content to decide what data is requested. See https://github.com/alpheios-project/lexis-cs/pull/1/files#r348649437 for the logic behind that.

So if message has a `getWords` object it means that a `getWords` action is requested. It could be possible to request several actions by on message (this is not implemented yet):
```
{
  getWord: {
     request params
  },
  getSomeOtherData: {
     request params
  }
}
```
It would request two actions `getWord` and `getSomeOtherData`. Results for these two actions will be returned in one response combined:
```
{
  getWord: {
     results
  },
  getSomeOtherData: {
     results
  }
}
```

I think it allows to simplify a transaction flow. Would such an approach make sense for us? What do you think?

Reply:
waiting on the sequence diagram before responding on this one :-)

Reply:
Here is how the diagram looks like:
![image](https://user-images.githubusercontent.com/18631055/69343649-6c945480-0c87-11ea-8706-97ec84514f6f.png)
Not sure if I set shapes of all participants correctly, but hope it will convey the idea.

Reply:
Here is a larger, more detailed diagram. Please click on it to open full screen in a separate window.

![image](https://user-images.githubusercontent.com/18631055/69722571-ef5a5b00-1130-11ea-810c-919882127fd2.png)

Hope it does explain what's going on better, but if not, please let me know and I'll add more details. Thanks!

Reply:
Yes this helps a lot thank you.

Let me try to state my understanding:

The Cedict service wraps an instance of a MessagingService. It runs in an Iframe
The MessagingService is instantiated during the DOMContentLoaded event on the iframe document
The Cedict service provides access to the Cedict data object, which encapsulates all the business logic of how to lookup words in the Cedict database.
A Messaging Service knows how to dispatch and receive messages to and from Destinations and keeps track of the request queue.
A Destination is responsible for listening for requests and sending a response. 
An WindowIFrameDestination is a type a Destination that uses the postMessage API to send and receive messages between an Iframe and its parent window.

Is this correct?

If so, then I think the only change I would request for now is that rather than the CEDICT configuration being defined as a constant of the WindowIFrameDestination module that it be defined as a constance of Cedict Service. Because the Cedict service encapsulates knowledge about the Cedict data object and how it can be addressed.  And because I think we might have other instances of WindowIFrameDestination for other types of data, used by other services and it would not be as flexible to have to register them all with the WindowIFrameDestination module.  Do you agree?